### ISSUE TITLE
Retouch replacement location influenced by crop

### ISSUE THREAD

**Issue by davidvj** [darktable-org/darktable#5766]
The location of the replacement area in being modified by the image crop.
It makes no difference if the cropping is done before or after retouch is done the replacement area is moved and does not correctly make the replacement.
I have seen this several times before and has often corrected in a next git version:
this is darktable 3.1.0+2466~ge328b89ea
I am using retouch in the copy mode and have tried several tools all with the same result. It is most noticeable then both the width and height are cropped.

**Screenshots**

**Platform (please complete the following information):**
 - Darktable Version: 
 - OS: 
 - 
 - 

**Additional context**

**Comment by elstoc** [comment]
I've seen this myself on-and-off but it's not consistent for me. I'll see the issue on an image, then restart or just exit back to the lighttable and it'll go away. It's never been consistent enough for me to be able to properly investigate. Every time I've tried to retrace my steps to figure out how it happened, it's been unreproducable.

Out of curiosity what happens if you drag the retouch module so that it runs before crop & rotate in the pipeline?

**Comment by davidvj** [comment]
Remind me please the shortcut for moving the modules ....

**Comment by ptilopteri** [comment]
<ctrl><shift><drag>

**Comment by davidvj** [comment]
Thanks for that keystroke reminder ....

Did a full system restart and everything now works perfectly *without* 
moving any of the modules. This is mimicking the problem that I have 
seen before .... it mostly works!

On this occasion I used both the perspective module (bi-directional) and 
the crop and rotate (chopping all 4 sides). Retouch is definitely after 
both of those modules. In retouch I applied various types of masks and 
masks were placed at differing zoom levels.

Next time I get the problem I will simply do a restart. hopefully 
somebody will eventually figure this out. (and I will still remember the 
fix!!)

David
